File: js_rails_final_project_walkthrough/dog_walker_backend/app/models/owner.rb
class Owner < ApplicationRecord
  # Include default devise modules. Others available are:
  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
  # devise :database_authenticatable, :registerable,
  #        :recoverable, :rememberable, :validatable

  has_many :dogs

  validates :name, presence: true
  validates :email, presence: true
  validates :address, presence: true

  devise :database_authenticatable,
          :jwt_authenticatable,
          :registerable,
          jwt_revocation_strategy: JwtBlacklist


end

File: js_rails_final_project_walkthrough/dogWalkerFrontend/src/components/owner.js
class Owner{
    constructor(owner){
        const { id, email, name, address, dogs } = owner
        this.id = id
        this.email = email
        this.name = name
        this.address = address
        this.dogs = dogs.map(d => new Dog(d))
    }


    get profileHTML(){
        return (`
            <h2>Welcome ${this.name}! </h2>
            <h3>Your dogs:</h3>
            <ul>
                ${this.dogs.map(d => d.liAndLinkHTML).join('')}
            </ul>
        `)
    }
}